iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/compaction/execute/recover/CompactionRecoverTask.java

```diff
@@ -274,28 +274,31 @@ private boolean checkIsTargetFilesComplete(
   private boolean deleteFile(TsFileIdentifier tsFileIdentifier) {
     boolean success = true;
     // delete tsfile
-    File file = tsFileIdentifier.getFileFromDataDirs();
-    if (!checkAndDeleteFile(file)) {
+    File tsFile = tsFileIdentifier.getFileFromDataDirs();
+    if (!checkAndDeleteFile(tsFile)) {
       success = false;
     }
 
     // delete resource file
-    file = getFileFromDataDirs(tsFileIdentifier.getFilePath() + TsFileResource.RESOURCE_SUFFIX);
-    if (!checkAndDeleteFile(file)) {
+    File resourceFile =
+        getFileFromDataDirs(tsFileIdentifier.getFilePath() + TsFileResource.RESOURCE_SUFFIX);
+    if (!checkAndDeleteFile(resourceFile)) {
       success = false;
     }
 
```

Comment:
Also check old version mods files and compaction mods files?

Reply:
Done